controllers/managedocs_controller.go

```diff
@@ -210,21 +217,29 @@ func (r *ManagedOCSReconciler) setStorageClusterParamsFromSecret(sc *ocsv1.Stora
 	if err != nil {
 		return fmt.Errorf("Invalid storage cluster size value: %v", sizeAsString)
 	}
-
 	r.Log.Info("Storage cluster parameters", "count", sdsCount)
 
 	var ds *ocsv1.StorageDeviceSet = nil
-	for _, item := range sc.Spec.StorageDeviceSets {
+	for index := range sc.Spec.StorageDeviceSets {
+		item := &sc.Spec.StorageDeviceSets[index]
 		if item.Name == deviceSetName {
-			ds = &item
+			ds = item
 			break
 		}
 	}
 	if ds == nil {
 		return fmt.Errorf("cloud not find default device set on stroage cluster")
 	}
 
-	sc.Spec.StorageDeviceSets[0].Count = sdsCount
+	// Prevent downscaling by comparing count from secret and count from storage cluster
+	if currSdsCount <= sdsCount {
+		ds.Count = sdsCount
+		r.Log.Info("Actual storage cluster parameter", "actualCount", ds.Count)
+	} else {
+		ds.Count = currSdsCount
+		r.Log.V(-1).Info("Cannot set storage device set count", "actualCount", ds.Count)
```

Comment:
This log warning does not convey the problem to the reader, please change to something like:

```suggestion
		r.Log.V(-1).Info("Requested storage device set count will result in a downscale, which is not supported. Skipping", "Reqested", sdsCount, "Current", ds.Count)
```